openfl/component/envoy/envoy.py

```diff
@@ -47,21 +47,36 @@ def run(self):
         while True:
             try:
                 # Workspace import should not be done by gRPC client!
-                experiment_name = self.director_client.get_experiment_data()
+                experiment_name = self.director_client.wait_experiment()
+                data_stream = self.director_client.get_experiment_data(experiment_name)
             except Exception as exc:
                 logger.error(f'Failed to get experiment: {exc}')
                 time.sleep(DEFAULT_RETRY_TIMEOUT_IN_SECONDS)
                 continue
+            data_file_path = self._save_data_stream_to_file(data_stream)
             self.is_experiment_running = True
             try:
-                self._run_collaborator(experiment_name)
+                with ExperimentWorkspace(
+                        experiment_name, data_file_path, is_install_requirements=True
+                ):
```

Comment:
Maybe rename `is_install_requirements` => `requirements` in order to avoid so long line.